pkg/limits/flags.go

```diff
@@ -0,0 +1,123 @@
+// This file and its contents are licensed under the Apache License 2.0.
+// Please see the included NOTICE for copyright information and
+// LICENSE for a copy of the license
+
+package limits
+
+import (
+	"flag"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/timescale/promscale/pkg/limits/mem"
+	"github.com/timescale/promscale/pkg/util"
+)
+
+var (
+	MemoryTargetMetric = prometheus.NewGauge(
```

Comment:
If we have just a single metric to be declared, I think we should use `promauto`.
https://pkg.go.dev/github.com/prometheus/client_golang/prometheus/promauto@v1.9.0

Reply:
if we switch to promauto we should do that across the entire codebase and in a separate PR? I think being consistent with the other stuff we have is better for now